packages/algoliasearch-helper/src/algoliasearch.helper.js

```diff
@@ -1389,6 +1427,81 @@ AlgoliaSearchHelper.prototype._search = function (options) {
   return undefined;
 };
 
+AlgoliaSearchHelper.prototype._recommend = function () {
+  var state = this.recommendState;
+  var queries = state.map((query) => ({ ...query, indexName: this.indexName }));
+
+  var queryId = this._recommendQueryId++;
+  this._currentNbRecommendQueries++;
+
+  try {
+    this.client
+      .getRecommendations(queries)
+      .then(this._dispatchRecommendResponse.bind(this, queryId))
+      .catch(this._dispatchRecommendError.bind(this, queryId));
+  } catch (error) {
+    // If we reach this part, we're in an internal error state
+    this.emit('recommendError', {
```

Comment:
```suggestion
    this.emit('recommend:error', {
```

this is possible I believe